qiskit/circuit/quantumcircuit.py

```diff
@@ -211,6 +217,48 @@ def __init__(self, *regs, name=None, global_phase=0, metadata=None):
             raise TypeError("Only a dictionary or None is accepted for circuit metadata")
         self._metadata = metadata
 
+    @property
+    def qregs(self):
+        """Returns a list of QuantumRegisters attached to the circuit."""
+
+        if self._reg_added_from_int and not QuantumCircuit._implicit_register_deprecation_shown:
+            warnings.warn(
+                "Initializing a QuantumCircuit with only the number of qubits and clbits previously "
+                'implicitly created a QuantumRegister("q") and optionally ClassicalRegister("c") '
+                "and added them to the circuit. This behavior has been deprecated and will be "
+                "removed in a future release. Instead, the circuit will be created with the correct "
+                "number of registerless qubits and clbits.",
+                DeprecationWarning,
```

Comment:
This likely isn't a `DeprecationWarning`, it's a `FutureWarning`; there's not going to be anything wrong in the future with doing `QuantumCircuit(2, 2).qregs`, it's just that the behaviour will have changed.

Having it be a `FutureWarning` will cause it to be shown to users in general, whereas deprecation warnings are hidden by default unless triggered by code in `__main__` (except see #7204, because Qiskit overrides that at the moment).